cores/esp32/esp32-hal-periman.c

```diff
@@ -33,9 +33,13 @@ bool perimanSetPinBus(uint8_t pin, peripheral_bus_type_t type, void * bus){
 		log_e("Bus is NULL");
 		return false;
 	}
+	if (type == ESP32_BUS_TYPE_INIT && bus != NULL){
+		log_e("Can't set a Bus to INIT Type");
+		return false;
+	}	
 	otype = pins[pin].type;
 	obus = pins[pin].bus;
-	if(type == otype && bus == obus){
+	if(type != ESP32_BUS_TYPE_INIT && type == otype && bus == obus){
```

Comment:
what's wrong in returning here if pin was already cleared? If the log message is a problem, you can print it only if type is not INIT

Reply:
I agree to just print the message if type is not INIT are return here.

Reply:
Changed to only print message when it is not INIT.